updated readme (#61)
sigp/lighthouse: README.md

This code-base is still very much under-development and does not provide any
user-facing functionality. For developers and researchers, there are several

> tests and benchmarks which may be of interest. Few basic setups needed before starting like:

Reviewer: ```suggestion
tests and benchmarks which may be of interest.

A few basic steps are needed to get set up:
```

Reviewer: This should be a separate paragraph as it describes a new idea.

>    1. Install [rustup](https://rustup.rs/).  It's a toolchain manager for Rust (Linux | macos | Windows). For installation run the below command in your terminal

Reviewer: ```suggestion
   1. Install [rustup](https://rustup.rs/).  It's a toolchain manager for Rust (Linux | macos | Windows). For simple one-line installation, you may run the below command in your terminal.
```

   ```
        $ curl https://sh.rustup.rs -sSf | sh
```

>    2. To configure your current shell run

Reviewer: ```suggestion
   2. To configure your current shell run:
```

  
   ```

>         $ source $HOME/.cargo/env

Reviewer: This does not exist for me if I installed a different way rather than using the one-liner. I had to add `~/.cargo/bin/` to my `$PATH` with:
`export PATH=$PATH:~/.cargo/bin`
Then I had to run `rustup install stable` and `rustup default stable` in order to download & select the rust setup.

We should make sure the instructions are generic enough to help people, even if they choose to install rustup from a different source (such as from the arch repositories, as in my case).

Author: @spble  Should I put one liner cmd ?

Reviewer 2: I think this is fine as it is, I'm going to resolve this in the name of expediting a solution.

We can always update this later if people start having issues. Arch people can figure it out themselves for now -- being on Arch I figure they like it that way :)

```
   

>    3. Use `rustup show` to get the info about the Rust setup. You will see active toolchain is in the stable version. 

Reviewer: ```suggestion
   3. Use the command `rustup show` to get information about the Rust installation. You should see that the active toolchain is the stable version. 
```

>    4. Run  `rustc --version` to get the version of rustup. For doing an update using` rustup update` .

Reviewer: ```suggestion
   4. Run  `rustc --version` to check the installation and version of rust.
      - Updates can be performed using` rustup update` .
```

   5. Navigate to the working directory.
   6. Run the test by using command `cargo test --all` . By running, it will pass all the required test cases. If you are doing it for the first time, then you can grab a coffee meantime. Usually, it takes time to build, compile and pass all test cases. If there is no error then, it means everything is working properly and it's time to get hand's dirty. In case, if there is an error, then please raise the [issue](https://github.com/sigp/lighthouse/issues).  We will help you.

>    7. The alternative of the above step, You  can also run benchmarks by using `cargo bench --all`

Reviewer: ```suggestion
   7. As an alternative to, or instead of the above step, you may also run benchmarks by using the command `cargo bench --all`
```
